ios/browser/api/favicon/favicon_loader.mm

```diff
@@ -0,0 +1,90 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/ios/browser/api/favicon/favicon_loader.h"
+#include "base/strings/sys_string_conversions.h"
+#include "brave/ios/browser/api/favicon/favicon_attributes.h"
+#include "components/favicon/core/large_icon_service.h"
+#include "components/favicon_base/favicon_types.h"
+#include "ios/chrome/browser/browser_state/chrome_browser_state.h"
+#import "ios/chrome/browser/favicon/favicon_loader.h"
+#include "ios/chrome/browser/favicon/favicon_service_factory.h"
+#include "ios/chrome/browser/favicon/ios_chrome_favicon_loader_factory.h"
+#include "ios/chrome/common/ui/favicon/favicon_attributes.h"
+#import "ios/chrome/common/ui/favicon/favicon_constants.h"
+#include "ios/chrome/common/ui/favicon/favicon_view.h"
+#import "net/base/mac/url_conversions.h"
+
+#if !defined(__has_feature) || !__has_feature(objc_arc)
+#error "This file requires ARC support."
+#endif
+
+// MARK: - Constants
+BraveFavIconLoaderSize const BraveFavIconLoaderSizeMin =
+    static_cast<NSInteger>(kMinFaviconSizePt);
+BraveFavIconLoaderSize const BraveFavIconLoaderSizeDesiredSmall =
+    static_cast<NSInteger>(kDesiredSmallFaviconSizePt);
+BraveFavIconLoaderSize const BraveFavIconLoaderSizeDesiredMedium =
+    static_cast<NSInteger>(kDesiredMediumFaviconSizePt);
+
+// MARK: - Implementation
+
+@interface BraveFavIconAttributes (Private)
+- (instancetype)initWithAttributes:(FaviconAttributes*)attributes;
+@end
+
+@interface BraveFavIconLoader () {
+  FaviconLoader* favicon_loader_;
+}
+@end
+
+@implementation BraveFavIconLoader
+- (instancetype)initWithBrowserState:(ChromeBrowserState*)mainBrowserState {
+  if ((self = [super init])) {
+    favicon_loader_ =
+        IOSChromeFaviconLoaderFactory::GetForBrowserState(mainBrowserState);
+    DCHECK(favicon_loader_);
+  }
+  return self;
+}
```

Comment:
We should be writing a KeyedServiceWrapper around this factory and using that in iOS, see https://github.com/brave/brave-core/blob/aff848ad16ccbe06b233bc84ccd6a2c4c97a1a8f/ios/browser/keyed_service/keyed_service_factory_wrapper.h#L22 and https://github.com/brave/brave-core/blob/master/ios/browser/brave_wallet/brave_wallet_factory_wrappers.h

Without it you are not providing the correct browser state during private browsing sessions